src/test/java/org/sonarsource/sonarlint/ls/connected/TaintVulnerabilitiesCacheTests.java

```diff
@@ -63,17 +65,27 @@ void testIssueConversion(String taintSource) {
     when(issue.getKey()).thenReturn("issueKey");
     when(issue.getFlows()).thenReturn(List.of(flow));
     when(issue.getSource()).thenReturn(taintSource);
-    when(issue.isOnNewCode()).thenReturn(true);
+    when(issue.isOnNewCode()).thenReturn(isOnNewCode);
 
-    var diagnostic = convert(issue).get();
+    var diagnostic = convert(issue, focusOnNewCode).get();
 
+    var severity = focusOnNewCode && !isOnNewCode ? DiagnosticSeverity.Hint : DiagnosticSeverity.Warning;
```

Comment:
IMHO this is duplicating the production code. I think it would be preferable to pass the "expected severity" as a test parameter.